zetaclient/bitcoin_client.go

```diff
@@ -870,9 +871,26 @@ func (ob *BitcoinChainClient) SelectUTXOs(amount float64, utxoCap uint8, nonce u
 		}
 	}
 	if total < amount {
-		return nil, 0, fmt.Errorf("SelectUTXOs: not enough btc in reserve - available : %v , tx amount : %v", total, amount)
+		return nil, 0, 0, 0, fmt.Errorf("SelectUTXOs: not enough btc in reserve - available : %v , tx amount : %v", total, amount)
+	}
+
+	// consolidate biggest possible UTXOs to maximize consolidated value
+	// consolidation happens only when there are more than 20 UTXOs
```

Comment:
Shouldn't we skip it then if `len(ob.utxos) < 20`?

Reply:
The number of iteration could be dynamic in future. For now, we can predict that each outTx produces 2 more UTXOs (new nonce-mark AND change to self) with the `2` hard coded. In the future, we'll produce more than 2 (a unpredictable number based on traffic) to improve the throughput. I'll leave it as a dynamic loop and it iterates at most 20 times which is negligible.
```
if len(ob.utxos)-len(results)+2 > int(consolidateRank) { // +2 for new nonce-mark and change to self
}
```